docs/native-modules-async.md

```diff
@@ -256,4 +256,146 @@ We've defined an `AsyncActionCompletedHandler` lambda and set it to be run when
 
 > **Important:** This example shows the minimum case, where you don't handle any errors within `GetHttpResponseAsync`, but you're not limited to this. You're free to detect error conditions within your code and call `capturedPromise.Reject()` yourself with (more useful) error messages at any time. However you should *always* include this final handler, to catch any unexpected and unhandled exceptions that may occur, especially when calling Windows APIs. Just be sure that you only call `Reject()` once and that nothing executes afterwards.
 
-That's it! If you want to see the complete `SimpleHttpModule`, see [`AsyncMethodExamples.h`](https://github.com/microsoft/react-native-windows-samples/blob/master/samples/NativeModuleSample/cppwinrt/windows/NativeModuleSample/AsyncMethodExamples.h).
\ No newline at end of file
+That's it! If you want to see the complete `SimpleHttpModule`, see [`AsyncMethodExamples.h`](https://github.com/microsoft/react-native-windows-samples/blob/master/samples/NativeModuleSample/cppwinrt/windows/NativeModuleSample/AsyncMethodExamples.h).
+
+## Executing calls to API on the UI thread
+
+Since version 0.64 the native module's code no longer runs on the UI thread. It means that each call to the API that should be executed on the UI thread now needs to be explicitly dispatched.
+
+To do that the [UIDispatcher](https://microsoft.github.io/react-native-windows/docs/IReactDispatcher) should be used.
+
+This part will cover the basic usage scenario of the `UIDispatcher` and its `Post()` method with the *FileOpenPicker* (for the explanation of opening files and folder with a picker on UWP please check the [Open files and folders with a picker](https://docs.microsoft.com/en-us/windows/uwp/files/quickstart-using-file-and-folder-pickers)).
+
+### Using `UIDispatcher` with C#
+
+Let's suppose we have a native module which opens a file using the *FileOpenPicker*.  
+Following the official example the native module's method launching the picker would look like the following:
+
+```cs
+  [ReactMethod("openFile")]
+  public async void OpenFile()
+  {
+    var picker = new Windows.Storage.Pickers.FileOpenPicker();
+    // Other initialization code
+    Windows.Storage.StorageFile file = await picker.PickSingleFileAsync();
+
+    if (file != null)
+    {
+      // File opened successfully
+    }
+    else
+    {
+      // Error while opening the file
+    }
+  }
+```
+However, since v0.64 this method would end up with `System.Exception: Invalid window handle`.
+So to avoid that, we need to wrap this call with the `UIDispatcher.Post` method.
+
+> **Note:** `UIDispatcher` is available via the `ReactContext`, which we can inject through a method marked as `ReactInitializer`:
+> ```cs
+>  [ReactInitializer]
+>  public void Initialize( ReactContext reactContext )
+>  {
+>      context = reactContext;
+>  }
+>```
+
+To do it, let's
+1. Separate the file's opening and handling from the UI thread logic, by moving this logic to the private method:

```

Comment:
this doesn't seem strictly necessary (esp if it's just calling one function), but might be a "good practice", depending.